homeassistant/components/bmw_connected_drive/button.py

```diff
@@ -0,0 +1,125 @@
+"""Support for BMW connected drive button entities."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from bimmer_connected.vehicle import ConnectedDriveVehicle
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import (
+    DOMAIN as BMW_DOMAIN,
+    BMWConnectedDriveAccount,
+    BMWConnectedDriveBaseEntity,
+)
+from .const import CONF_ACCOUNT, DATA_ENTRIES
+
+
+@dataclass
+class BMWButtonEntityDescription(ButtonEntityDescription):
+    """Class describing BMW button entities."""
+
+    enabled_when_read_only: bool = False
+    remote_function: str | None = None
+    account_function: str | None = None
+
+
+BUTTON_TYPES: tuple[BMWButtonEntityDescription, ...] = (
+    BMWButtonEntityDescription(
+        key="light_flash",
+        icon="mdi:car-light-alert",
+        name="Flash Lights",
+        remote_function="trigger_remote_light_flash",
+    ),
+    BMWButtonEntityDescription(
+        key="sound_horn",
+        icon="mdi:bullhorn",
+        name="Sound Horn",
+        remote_function="trigger_remote_horn",
+    ),
+    BMWButtonEntityDescription(
+        key="activate_air_conditioning",
+        icon="mdi:hvac",
+        name="Activate Air Conditioning",
+        remote_function="trigger_remote_air_conditioning",
+    ),
+    BMWButtonEntityDescription(
+        key="deactivate_air_conditioning",
+        icon="mdi:hvac-off",
+        name="Deactivate Air Conditioning",
+        remote_function="trigger_remote_air_conditioning_stop",
+    ),
+    BMWButtonEntityDescription(
+        key="find_vehicle",
+        icon="mdi:crosshairs-question",
+        name="Find Vehicle",
+        remote_function="trigger_remote_vehicle_finder",
+    ),
+    BMWButtonEntityDescription(
+        key="refresh",
+        icon="mdi:refresh",
+        name="Refresh from API",
```

Comment:
Renaming to "Refresh data from BMW API" might make the function of this button slightly more clear.

Reply:
Thats true. Would you prefer "Refresh from BMW API" or "Refresh from BMW servers"?

Reply:
For Nissan Leaf and Renault the docs mention 'cloud service', because there's not an official API.

Reply:
`cloud` does make sense in this case!